Drive/rover_drive_system.hpp

```diff
@@ -99,76 +176,165 @@ class RoverDriveSystem
+        sjsu::LogInfo("Translating...");
+        HandleTranslationMode(speed, angle);
+        break;
+      default:
+        SetWheelSpeed(kZeroSpeed);
+        sjsu::LogError("Unable to assign drive mode handler!");
+        break;
     }
-    if (current_mode_ == Mode::kSpin)
+    current_speed_ = speed;
+  };
+
+  /// Sets the new driving mode for the rover. Rover will stop before switching
+  /// @param mode Three Modes: D (drive), S (spin), T (translation)
+  void SetMode(char mode = 'S')
+  {
+    switch (mode)
     {
-      HandleSpinMode();
-    }
-    if (current_mode_ == Mode::kTranslation)
+      case 'D':
+      case 'd':
+        current_mode_ = Mode::kDrive;
+        SetWheelSpeed(kZeroSpeed);
+        SetDriveMode();
+        sjsu::LogInfo("Drive mode set");
+        break;
+      case 'S':
+      case 's':
+        current_mode_ = Mode::kSpin;
+        SetWheelSpeed(kZeroSpeed);
+        SetSpinMode();
+        sjsu::LogInfo("Spin mode set");
+        break;
+      case 'T':
+      case 't':
+        current_mode_ = Mode::kTranslation;
+        SetWheelSpeed(kZeroSpeed);
+        SetTranslationMode();
+        sjsu::LogInfo("Translation mode set");
+        break;
+      default:
+        SetWheelSpeed(kZeroSpeed);
+        sjsu::LogError("Unable to assign drive mode!");
+    };
+  };
+
+  /// Handles the rover movement depending on the mode
+  /// @param rotation_angle adjusts wheel position depending on mode.
+  /// @param wheel_speed adjusts the movement speed of the rover.
+  void HandleRoverMovement(float roatation_angle, float wheel_speed)
+  {
+    units::angle::degree_t angle(roatation_angle);
+    units::angular_velocity::revolutions_per_minute_t speed(wheel_speed);
+    switch (current_mode_)
     {
-      HandleTranslationMode();
+      case Mode::kDrive:
+        sjsu::LogInfo("Driving...");
+        HandleDriveMode(speed, angle);
+        break;
+      case Mode::kSpin:
+        sjsu::LogInfo("Spining...");
+        HandleSpinMode(speed);
+        break;
+      case Mode::kTranslation:
+        sjsu::LogInfo("Translating...");
+        HandleTranslationMode(speed, angle);
+        break;
+      default:
+        SetWheelSpeed(kZeroSpeed);
+        sjsu::LogError("Unable to assign drive mode handler!");
+        break;
     }
+    current_speed_ = speed;
   };
 
-  /// Calculates position to set each of the rover wheels to initially
-  void SetDriveMode();
-  void SetSpinMode();
-  void SetTranslationMode();
+  // ======================
+  // = DRIVE MODE SETTERS =
+  // ======================
 
-  /// Handles the movement for their respective modes.
-  void HandleDriveMode();
-  void HandleSpinMode();
-  void HandleTranslationMode();
+  /// Aligns rover wheels all in the same direction, facing foward
+  void SetDriveMode()
+  {
+    units::angle::degree_t left_wheel_angle  = -45_deg;
+    units::angle::degree_t right_wheel_angle = -135_deg;
+    units::angle::degree_t back_wheel_angle  = 90_deg;
```

Comment:
You can make these constexpr since they aren't changing.

Reply:
didn't like constexpr so I assigned them as const